model_compression_toolkit/core/common/pruning/importance_metrics/lfh_importance_metric.py

```diff
@@ -120,22 +120,21 @@ def _get_entry_node_to_score(self, entry_nodes: List[BaseNode]) -> Dict[BaseNode
         """
 
         # Initialize HessianInfoService for score computation.
+
         hessian_info_service = HessianInfoService(graph=self.float_graph,
-                                                  representative_dataset_gen=self.representative_data_gen,
                                                   fw_impl=self.fw_impl)
 
         # Fetch and process Hessian scores for output channels of entry nodes.
-        nodes_scores = []
-        for node in entry_nodes:
```

Comment:
Yes, I though we talked about this, but maybe only I thought that :) I can restore it. Gradient-wise it should be identical but there is a small difference that now the same random vector is used for all nodes whereas previously random vector was generated for each node.

Reply:
Yes you are correct that it should be identical, and also that the small change due to randomness is probably negligible, but I prefer to avoid changing unrelated algorithms to prevent unexpected issues, this is already a big enough refactor that affects many modules

Reply:
restored